test/abbrtypes.ml

```diff
@@ -0,0 +1,73 @@
+(* Slacko's public interface doesn't let us easily construct Slacko.user and
```

Comment:
Why do you need to extract values from these types? To do comparisons? I feel this is a bit brittle, as the types now have to be declared effectively twice. Correct me if I am misunderstanding.

Another idea: adding comparison functions to Slacko which compare some (relevant) values of the records without exposing the inner workings.